### ISSUE TITLE
MSBuild Server

### ISSUE THREAD

**Issue by rokonec** [dotnet/msbuild#6392]
To improve performance of consequent MSBuild execution keep 'Entry MSBuild node' process alive and connect to it when there is a new build requests.

In first version lets use existing MSBULDNOINPROCNODE escape hatch (blocked by #3517).

Final version is yet TBD, but our current thoughts about it are as follows:
- Entry MSBuild node will host named pipe connection end point.
- CLI checks existence and busy/ready state of Entry MSBuild node by named mutexes
- CLI starts MSBuild entry node if not running
- CLI will connect to running MSBuild entry node and pass it command line arguments
- CLI fallbacks to current behavior (in process MSBuild dll) if entry node is busy
- MSBuild entry node process will have floating window TTL of (by default) 15 minutes

This approach will be more beneficial after we implement some kind of preheating (speculative processes start, prebuild, ...). If preheating will not make it into 6.0 we might consider to start Entry MSBuild node in background while compiling by CLI in process loaded MSBuild. This way we will not pay as big price in the very-cold scenarios.

- [x] MSBuild Server V1 #6494
- [x] Able to start MSBuild node as Server #6424
- [x] Define and handle build request message #6425
- [x] Forward console output to CLI #6466
- [x] Create fully functional prototype and measure it #6467
- [x] dotnet SDK CLI integration #6426
- [ ] Manual exploratory testing and measurement #6427

**Comment by rokonec** [comment]
Feature has been parked into: https://github.com/dotnet/msbuild/tree/feature/msbuild-server-v2--parked

See linked issues for more detail. Works left is estimated to about 3 week/person.
